src/toil/provisioners/abstractProvisioner.py

```diff
@@ -41,10 +41,17 @@ def __eq__(self, other):
 
     def __lt__(self, other):
         return self.preemptable > other.preemptable or \
-            self.memory < other.memory or \
-            self.cores < other.cores or \
-            self.disk < other.disk or \
-            self.wallTime < other.wallTime
+               self.memory < other.memory or \
+               self.cores < other.cores or \
+               self.disk < other.disk or \
+               self.wallTime < other.wallTime
+
+    def __gt__(self, other):
```

Comment:
You shouldn't need `__gt__` here, at least if this weird multiple-inheritance crap does its job. The `functools.total_ordering` decorator defines all seven comparison methods for you consistently, as long as you define `__eq__` and one of <=, <, >, >=. The point of this weird Ordering stuff is to avoid defining all seven, because this is super error-prone and a flipped sign can lead to bugs. Did something come up which meant `__gt__` needed to be defined?

Reply:
also, pep8 recommends putting multiline expressions like this in parenthesis rather than backslash escaping each line.  just an FYI, not a requested change, but it does improve readability IMUO.

Reply:
You're right, I misunderstood originally.  I used it temporarily, and saw the `@total_ordering` decorator later but don't really have a good reason for leaving it in.  Will remove it when I push the next few commits.  It's taking a bit for me to absorb all of the inheritance throughout.